6.4 C - Sunday August 9, 2015

cbotwell opened the pull request:
:open_file_folder: Starting Branch C

cbotwell committed:
added model and collection

cbotwell committed:
added formview and hbs for form

cbotwell committed:
got submit form working

cbotwell committed:
added linkview and template

cbotwell committed:
added tagview and template, need to filter for dupes

cbotwell committed:
working on filtering tags

rtablada commented on assets/js/views/tagview.js:
```diff
@@ -0,0 +1,27 @@
+    var _this = this;
+
+    this.collection.fetch().then(function() {
+      var html = _this.template(_.filter(_this.collection.models, function(model) {
```
Instead of having to do all of this work around to get `_.filter` and `collection.models`: backbone gives you all of the array functions from underscore and makes them available on a collection: http://backbonejs.org/#Collection-Underscore-Methods

cbotwell committed:
started on router, added scss

cbotwell committed:
more styles

cbotwell committed:
color style

cbotwell committed:
styles

cbotwell committed:
changed color

cbotwell committed:
flipped color

cbotwell committed:
working on router

cbotwell committed:
continuing work on router

cbotwell committed:
working on router

cbotwell commented:
Still having trouble filtering, render not seeming to work either.

cbotwell commented:
With help from Ryan: 
- changed the router to have an optional tag
- Added backbone history start
- Updated filterLinksByTag 
  - to return only models with matching tags if one is provided 
  - otherwise return all models
  - then used map to convert models to JSON
- Updated TagView Render
  - filtered tags variable
  - used reduce with a starting link of all tags
  - checks if the array has an object with a name that matches the tag
  - if not it pushes an new object with a name: tag and path: tag
- Updated the taglist.hbs based on the new array with names and paths
- Don't need the LinkView anymore because the router returns all links when the tag doesn't exist

cbotwell committed:
got router and tag filter working

rtablada commented on assets/js/routers/router.js:
```diff
@@ -0,0 +1,28 @@
+  },
+
+  initialize: function() {
+    this.template = AppTemplates.linkview;
```
Please use views for all templating. The router will be unable to handle any events that may need to fire and is pretty poor at listening for underlying events on collections or models. If it helps you can `bower install --save marionette` and  lean on Marionette to handle rendering and/or regions.
